libs/wingsdk/src/shared-gcp/bucket.inflight.ts

```diff
@@ -211,13 +212,47 @@ export class BucketClient implements IBucketClient {
     return `https://storage.googleapis.com/${this.bucketName}/${key}`;
   }
 
-  // TODO: implement signedUrl
-  // https://github.com/winglang/wing/issues/4599
-
+  /**
+   * Returns a presigned URL for the specified key in the bucket.
+   * @param key The key of the object in the bucket.
+   * @param opts The options including the action and the duration for the signed URL.
+   * @returns The presigned URL string.
+   * @inflight
+   */
   public async signedUrl(
-    _key: string,
-    _options?: BucketSignedUrlOptions
+    key: string,
+    opts: BucketSignedUrlOptions
   ): Promise<string> {
-    throw new Error("Method not implemented.");
+    const gcsFile = this.bucket.file(key);
+    let gcsAction: "read" | "write" | "delete" | "resumable";
```

Comment:
is there an enum for that?

Reply:
I checked for it, but that's just how it's defined sadly.

![image](https://github.com/winglang/wing/assets/10464497/71fb7668-3455-4197-b7f3-742a612c4c3d)